ambry-clustermap/src/main/java/com/github/ambry/clustermap/MySqlReportAggregatorTask.java

```diff
@@ -0,0 +1,134 @@
+      AccountStatsStore accountStatsStore, Callback<StatsSnapshot> callback, ClusterMapConfig clusterMapConfig,
+      Time time) {
+    this.manager = manager;
+    clusterAggregator = new HelixClusterAggregator(relevantTimePeriodInMs);
+    this.statsReportType = statsReportType;
+    this.accountStatsStore = accountStatsStore;
+    this.callback = callback;
+    this.clusterMapConfig = clusterMapConfig;
+    this.time = time;
+  }
+
+  @Override
+  public TaskResult run() {
+    Pair<StatsSnapshot, StatsSnapshot> results = null;
+    Exception exception = null;
+    try {
+      List<String> instanceNames = manager.getClusterManagmentTool().getInstancesInCluster(manager.getClusterName());
+      Map<String, StatsWrapper> statsWrappers = new HashMap<>();
+      for (String instanceName : instanceNames) {
+        // Helix instance name would suffix port number, here let's take port number out.
+        instanceName = stripPortNumber(instanceName);
+        statsWrappers.put(instanceName,
+            accountStatsStore.queryStatsOf(clusterMapConfig.clusterMapClusterName, instanceName));
+      }
+      results = clusterAggregator.doWorkOnStatsWrapperMap(statsWrappers, statsReportType);
+      accountStatsStore.storeAggregatedStats(results.getSecond());
+      // Create a base report at the beginning of each month.
+      // Check if there is a base report for this month or not.
+      if (clusterMapConfig.clustermapEnableAggregatedMonthlyAccountReport
+          && statsReportType == StatsReportType.ACCOUNT_REPORT) {
+        // Get the month, if not the same month, then copy the aggregated stats and update the month
+        String currentMonthValue =
+            LocalDateTime.ofEpochSecond(time.seconds(), 0, zoneOffset).format(TIMESTAMP_FORMATTER);
+        String recordedMonthValue = accountStatsStore.queryRecordedMonth(clusterMapConfig.clusterMapClusterName);
+        if (recordedMonthValue == null || recordedMonthValue.isEmpty() || !currentMonthValue.equals(
+            recordedMonthValue)) {
+          accountStatsStore.takeSnapshotOfAggregatedStatsSetMonth(clusterMapConfig.clusterMapClusterName,
+              currentMonthValue);
+        }
+      }
+      return new TaskResult(TaskResult.Status.COMPLETED, "Aggregation success");
+    } catch (Exception e) {
+      logger.error("Exception thrown while aggregating stats from health reports across all nodes ", e);
+      exception = e;
+      return new TaskResult(TaskResult.Status.FAILED, "Exception thrown");
+    } finally {
+      if (clusterMapConfig.clustermapEnableContainerDeletionAggregation && callback != null && results != null
+          && statsReportType.equals(StatsReportType.ACCOUNT_REPORT)) {
+        callback.onCompletion(results.getFirst(), exception);
+      }
+    }
+  }
+
+  private String stripPortNumber(String instanceName) {
+    return instanceName.replace("_" + clusterMapConfig.clusterMapPort, "");
```

Comment:
this may not be 100% accurate for a local deployment where multiple processes share the same host name but different ports.

Reply:
This method is just trying to get the hostname. I will append the port number later in the database method. So in helix the  instance name looks something like this `ltx1-app12345.prod.linkedin.com_15999`, but in the database, the hostname looks like `ltx1-app12345.prod_15999`. So I have to get the hostname first, remove the domain name and then append the port number. Even if we have several processes running at the same host, as long as their port numbers are different, it will be fine for database.

Reply:
This seems like this would have problems removing the port number if different hosts in the cluster have different ports. For example, let's say the cluster has `host1_1234` and `host2_2345` and `clusterMapConfig.clusterMapPort` is set to 1234. Then, this method would work correctly for host1, but return `host2_2345` for host2.
Maybe it is better to split by underscore and take the first part of it.

Reply:
But in different ambry server process, the port number in clustermap should be the port number that this process registered with helix cluster. If the cluster has `host1_1234` and `host2_2345`, then the clustermap port number in `host1` would be `1234` and the clustermap port number in `host2` would be `2345`.

Reply:
Maybe I am misreading this, but it seems like `stripPortNumber` is called for all hosts in a cluster (line 86), but `clusterMapConfig.clusterMapPort` is static and only represents this host's port number.

Reply:
oh, you are absolutely right, I was just being stupid, will update.